turbopack/crates/turbo-tasks/src/id_factory.rs

```diff
@@ -12,13 +12,19 @@ use concurrent_queue::ConcurrentQueue;
 /// For ids that may be re-used, see [`IdFactoryWithReuse`].
 pub struct IdFactory<T> {
     next_id: AtomicU64,
+    max_id: u64,
     _phantom_data: PhantomData<T>,
 }
 
 impl<T> IdFactory<T> {
     pub const fn new() -> Self {
+        Self::new_with_range(1, u32::MAX as u64)
```

Comment:
I've been using `IdFactory` with `ExecutionId`, which is a u64: https://github.com/vercel/next.js/blob/68f8c9746151438335434a8cf7eebbe373b3670b/turbopack/crates/turbo-tasks/src/id.rs#L72

We should use `u64::MAX` here as the default, and allow the overflow detection for `u32` to happen when we `try_into` from u64 to the generic type `T` later.

Reply:
Should we remove the constructor without arguments and enforce passing a max id?

Reply:
Yeah, that's reasonable.